docs/examples/101/databricks-all-in-one-template-for-vnet-injection/main.bicep

```diff
@@ -1,76 +1,181 @@
+          destinationAddressPrefix: 'Sql'
+          access: 'Allow'
+          priority: 101
+          direction: 'Outbound'
+        }
+      }
+      {
+        name: 'Microsoft.Databricks-workspaces_UseOnly_databricks-worker-to-storage'
+        properties: {
+          description: 'Required for workers communication with Azure Storage services.'
+          protocol: 'Tcp'
+          sourcePortRange: '*'
+          destinationPortRange: '443'
+          sourceAddressPrefix: 'VirtualNetwork'
+          destinationAddressPrefix: 'Storage'
+          access: 'Allow'
+          priority: 102
+          direction: 'Outbound'
+        }
+      }
+      {
+        name: 'Microsoft.Databricks-workspaces_UseOnly_databricks-worker-to-worker-outbound'
+        properties: {
+          description: 'Required for worker nodes communication within a cluster.'
+          protocol: '*'
+          sourcePortRange: '*'
+          destinationPortRange: '*'
+          sourceAddressPrefix: 'VirtualNetwork'
+          destinationAddressPrefix: 'VirtualNetwork'
+          access: 'Allow'
+          priority: 103
+          direction: 'Outbound'
+        }
+      }
+      {
+        name: 'Microsoft.Databricks-workspaces_UseOnly_databricks-worker-to-eventhub'
+        properties: {
+          description: 'Required for worker communication with Azure Eventhub services.'
+          protocol: 'Tcp'
+          sourcePortRange: '*'
+          destinationPortRange: '9093'
+          sourceAddressPrefix: 'VirtualNetwork'
+          destinationAddressPrefix: 'EventHub'
+          access: 'Allow'
+          priority: 104
+          direction: 'Outbound'
         }
       }
     ]
   }
 }
 
-resource nsg 'Microsoft.Network/networkSecurityGroups@2020-06-01' = {
-  name: nsgName
+resource vnet 'Microsoft.Network/virtualNetworks@2020-05-01' = {
   location: location
-}
-
-resource vnet 'Microsoft.Network/virtualNetworks@2020-06-01' = {
   name: vnetName
-  location: location
   properties: {
     addressSpace: {
       addressPrefixes: [
         vnetCidr
       ]
     }
     subnets: [
-      publicSubnet
-      privateSubnet
+      {
+        name: publicSubnetName
+        properties: {
+          addressPrefix: publicSubnetCidr
+          networkSecurityGroup: {
+            id: nsg.id
+          }
+          delegations: [
+            {
+              name: 'databricks-del-public'
+              properties: {
+                serviceName: 'Microsoft.Databricks/workspaces'
+              }
+            }
+          ]
+        }
+      }
+      {
+        name: privateSubnetName
+        properties: {
+          addressPrefix: privateSubnetCidr
+          networkSecurityGroup: {
+            id: nsg.id
+          }
+          delegations: [
+            {
+              name: 'databricks-del-private'
+              properties: {
+                serviceName: 'Microsoft.Databricks/workspaces'
+              }
+            }
+          ]
+        }
+      }
```

Comment:
Is the following TODO still relevant?

resource ws 'Microsoft.Databricks/workspaces@2018-04-01' = {
  name: workspaceName
  location: location
  sku: {
    name: pricingTier
  }
  properties: {
    // TODO: improve once we have scoping functions
    managedResourceGroupId: managedResourceGroupId